snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -45,6 +59,10 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.destroy()
 
+    @abc.abstractproperty
+    def _snaps_path_or_dev(self) -> str:
```

Comment:
Can you give me of an example for when a device would be used here instead of just a path on the host? This method name doesn't seem clear to me. Given that it corresponds to the `_SNAPS_MOUNTPOINT` variable above, perhaps something like `_snaps_mount_root` or something would be more suitable?

Reply:
in the case of qemu it would be a 9p created device instead of an actual path.
`_snaps_mount_root` as a property instead of this one?